zubhub_backend/zubhub/activities/models.py

```diff
@@ -0,0 +1,161 @@
+                                 related_name="activities")
+    video = models.URLField(max_length=1000, blank=True, null=True)
+    materials_used = models.CharField(max_length=5000)
+    materials_used_image = models.ForeignKey(Image,
+                                on_delete=models.CASCADE,
+                                null=True,
+                                related_name="image",
+                                blank=True)
+    inspiring_artist = models.ForeignKey(InspiringArtist,
+                                on_delete=models.CASCADE,
+                                null=True,
+                                related_name="inspiring_artist",
+                                blank=True)
+    views = models.ManyToManyField(Creator,
+                                   blank=True,
+                                   related_name="activities_viewed")
+    views_count = models.IntegerField(blank=True, default=0)
+    saved_count = models.IntegerField(blank=True, default=0)
+    saved_by = models.ManyToManyField(Creator,
+                                      blank=True,
+                                      related_name="saved_for_future")
+    created_on = models.DateTimeField(default=timezone.now)
+    publish = models.models.BooleanField(default=False)
+
+    def save(self, *args, **kwargs):
+        
+        if self.slug:
+            pass
+        else:
+            uid = str(uuid.uuid4())
+            uid = uid[0:floor(len(uid) / 6)]
+            self.slug = slugify(self.title) + "-" + uid
+            
+        super().save(*args, **kwargs)
+
+    def __str__(self):
+        return self.title
+    
+
+class InspiringExamples(models.Model):
+    activity = models.ForeignKey(Activity,
+                                on_delete=models.CASCADE,
+                                null=True,
+                                related_name="activity",
+                                blank=True)
+    description = models.CharField(max_length=10000, null=True, blank=True)
+    credit = models.CharField(max_length=1000, null=True, blank=True)
+    image = models.ForeignKey(Image,
+                                on_delete=models.CASCADE,
+                                null=True,
+                                related_name="image",
+                                blank=True)
+
+    def __str__(self):
+        return self.image.image_url
+
+class ActivityImages(models.Model):
+    activity = models.ForeignKey(Activity,
+                                on_delete=models.CASCADE,
+                                null=True,
+                                related_name="activity",
+                                blank=True)
+    image = models.ForeignKey(Image,
+                                on_delete=models.CASCADE,
+                                null=True,
+                                related_name="image",
+                                blank=True)
+
+    def __str__(self):
+        return self.image.image_url       
```

Comment:
we might need to specify which image is default to display as a demo image if specified by user "this is not handled in the frontend yet " what do you suggest ? we add default boolean field